src/hooks/useZodForm.ts

```
@@ -0,0 +1,25 @@
+import { DefaultValues, useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { UseMutateFunction } from '@tanstack/react-query';
+import { z } from 'zod';
+
+export const useZodForm = <T extends z.ZodType>(
+  schema: T,
+  mutation: UseMutateFunction<any, Error, z.infer<T>, unknown>,
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)